crates/kgraph_utils/src/mca.rs

```diff
@@ -257,17 +262,229 @@ fn compile_payment_method_enabled(
+        cgraph::Relation::Positive,
+        cgraph::Strength::Normal,
+    ));
+    if let Some(country) = config.country.clone() {
+        let node_country = country
+            .into_iter()
+            .map(api_enums::Country::from_alpha2)
+            .map(IntoDirValue::into_dir_value)
+            .collect::<Result<Vec<_>, _>>()?;
+        let country_agg = builder
+            .make_in_aggregator(node_country, Some("Configs for Country"), None::<()>)
+            .map_err(KgraphError::GraphConstructionError)?;
+        agg_nodes.push((
+            country_agg,
+            cgraph::Relation::Positive,
+            cgraph::Strength::Weak,
+        ))
+    }
+
+    if let Some(currency) = config.currency.clone() {
+        let node_currency = currency
+            .into_iter()
+            .map(IntoDirValue::into_dir_value)
+            .collect::<Result<Vec<_>, _>>()?;
+        let currency_agg = builder
+            .make_in_aggregator(node_currency, Some("Configs for Currency"), None::<()>)
+            .map_err(KgraphError::GraphConstructionError)?;
+        agg_nodes.push((
+            currency_agg,
+            cgraph::Relation::Positive,
+            cgraph::Strength::Normal,
+        ))
+    }
+    if let Some(capture_method) = config
+        .not_available_flows
+        .and_then(|naf| naf.capture_method)
+    {
+        let make_capture_node = builder.make_value_node(
+            cgraph::NodeValue::Value(dir::DirValue::CaptureMethod(capture_method)),
+            Some("Configs for CaptureMethod"),
+            None::<()>,
+        );
+        agg_nodes.push((
+            make_capture_node,
+            cgraph::Relation::Negative,
+            cgraph::Strength::Normal,
+        ))
+    }
+
+    builder
+        .make_all_aggregator(
+            &agg_nodes,
+            Some("Country & Currency Configs With Payment Method Type"),
+            None::<()>,
+            None,
+        )
+        .map_err(KgraphError::GraphConstructionError)
+}
+fn compile_config_graph(
+    builder: &mut cgraph::ConstraintGraphBuilder<'_, dir::DirValue>,
+    config: &CountryCurrencyFilter,
+    connector: &api_enums::RoutableConnectors,
+) -> Result<cgraph::NodeId, KgraphError> {
+    let mut agg_node_id: Vec<(cgraph::NodeId, cgraph::Relation, cgraph::Strength)> = Vec::new();
+    let mut pmt_enabled: Vec<dir::DirValue> = Vec::new();
+    if let Some(pmt) = config
+        .connector_configs
+        .get(connector)
+        .or(config.default_configs.as_ref())
+        .map(|inner| inner.0.clone())
+    {
+        for key in pmt.keys().cloned() {
+            match key {
+                PaymentMethodFilterKey::PaymentMethodType(pm) => {
+                    let dir_val_pm = pm.into_dir_value().map(Into::into)?;
+                    pmt_enabled.push(dir_val_pm);
+
+                    let curr = pmt
+                        .get(&PaymentMethodFilterKey::PaymentMethodType(pm))
+                        .map(|country| {
```

Comment:
Just iterate over both keys and values, instead of iterating over keys, and retrieving an optional value inside the loop (removes the need for handling optional values).